mesa/visualization/components/altair_components.py

```diff
@@ -67,93 +198,107 @@ def SpaceAltair(
     # Apply post-processing if provided
     if post_process is not None:
         chart = post_process(chart)
+
     solara.FigureAltair(chart)
 
 
-def _get_agent_data_old__discrete_space(space, agent_portrayal):
-    """Format agent portrayal data for old-style discrete spaces.
+def axial_to_pixel(q, r, size=1):
+    """Convert axial coordinates (q, r) to pixel coordinates for hexagonal grid."""
+    x = size * math.sqrt(3) * (q + r / 2)
+    y = size * 1.5 * r
+    return x, y
 
-    Args:
-        space: the mesa.space._Grid instance
-        agent_portrayal: the agent portrayal callable
 
-    Returns:
-        list of dicts
-
-    """
-    all_agent_data = []
-    for content, (x, y) in space.coord_iter():
-        if not content:
-            continue
-        if not hasattr(content, "__iter__"):
-            # Is a single grid
-            content = [content]  # noqa: PLW2901
-        for agent in content:
-            # use all data from agent portrayal, and add x,y coordinates
-            agent_data = agent_portrayal(agent)
-            agent_data["x"] = x
-            agent_data["y"] = y
-            all_agent_data.append(agent_data)
-    return all_agent_data
-
-
-def _get_agent_data_new_discrete_space(space: DiscreteSpace, agent_portrayal):
-    """Format agent portrayal data for new-style discrete spaces.
+def get_agent_data(space, agent_portrayal):
```

Comment:
For matplotlib, we use a slightly different approach in particular for networks and hexgrids: we update the location inside the respective plotting functions. 

Would it be possible to have a single overarching get_agent_data method that is valid for both matplotlib and altair?